credentials/credentials.go

```diff
@@ -77,23 +104,81 @@ func GetNewRelicLicenseKey(ctx context.Context, conf *config.Configuration) (str
 		return conf.LicenseKey, nil
 	}
 
+	var err error
+	var licenseKey string
+
 	secretId := getLicenseKeySecretId(conf)
+	if secretId != "" {
+		licenseKey, err = tryLicenseKeyFromSecret(ctx, secretId)
+		if err == nil {
+			return licenseKey, nil
+		}
+	}
+
+	parameterName := getLicenseKeyParameterName(conf)
+	if parameterName != "" {
+		licenseKey, err = tryLicenseKeyFromParameter(ctx, parameterName)
+		if err == nil {
+			return licenseKey, nil
+		}
+	}
+
+	envLicenseKey, found := os.LookupEnv(defaultSecretId)
+	if found {
+		return envLicenseKey, nil
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	util.Logln("No configured license key found, attempting fallbacks")
```

Comment:
I think my concern with this new approach is there's a lot of stuff that can silently fail. With the sanity checks in place is it even possible to get into a state where you have more than one source configured?

Also, trying to lookup a secret and parameter with the name `NEW_RELIC_LICENSE_KEY` seems new? Probably we can leave that out?

Reply:
Yeah, I definitely wouldn't do it this way if this was internal code, but I'm trying to avoid refactoring as much as possible.

Reply:
I'm not actually sure if the sanity check prevents the lambda from running, or just prints an error.

However, I don't think the silent failures are a problem here:
- if a secret and parameter are both configured and both lookups fail, we'll still return the second failure.
- if the fallback `NEW_RELIC_LICENSE_KEY` lookups fail, we'll still return a error.

Reply:
Maybe just being more explicit via logs when falling back / an error fetching a key is returned? That'd probably be helpful to someone trying to debug why their key is not being located.

Reply:
I'm not quite sure what you're suggesting. This is what the logs look like now:
```
[NR_EXT] New Relic Lambda Extension starting up
[NR_EXT] Fetching license key from parameter name /sit/global/newrelic/license_key_does_not_exist
[NR_EXT] Failed to retrieve New Relic license key ParameterNotFound: 
[NR_EXT] Starting no-op mode, no telemetry will be sent
```